### PR TITLE
6371 migrate sharding tags par2

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**
#245
**Description**

- implement add & edit entrypoint mapping dialog
- and some improvement and fix 

**Additional context**

### CODE REVIEW

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-diaupewyln.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags-par2/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-svmcwwoyhm.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags-par2/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-mapping-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  depending on API tags.
+</gio-banner-info>
+
+<form autocomplete="off" [formGroup]="mappingForm" (ngSubmit)="onSubmit()">
+  <mat-dialog-content>
+    <div class="form">
+      <mat-form-field appearance="fill">
+        <mat-label>Entrypoint Url</mat-label>
+        <input matInput formControlName="value" type="url" placeholder="https://api.mycompany.com" required />
+        <mat-error *ngIf="mappingForm.get('value').hasError('required')">Entrypoint Url is required.</mat-error>
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-mapping-dialog.component.html:31]:
```suggestion
        <mat-error *ngIf="mappingForm.get('value').hasError('required')">Entrypoint url is required.</mat-error>
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts [hunk 2]
```diff
@@ -259,11 +260,55 @@ export class OrgSettingsTagsComponent implements OnInit, OnDestroy {
...
+          entrypoint: this.entrypoints.find((e) => e.id === entrypoint.id),
+        },
+        role: 'dialog',
+        id: 'addTagDialog',
+      })
+      .afterClosed()
+      .pipe(
+        takeUntil(this.unsubscribe$),
+        filter((result) => !!result),
+        switchMap((newEntrypoint) => this.entrypointService.create(newEntrypoint)),
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]: `.create(`  here too and not `.update`?

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts [hunk 3]
```diff
@@ -210,14 +210,27 @@ export class OrgSettingsTagsComponent implements OnInit, OnDestroy {
...
-          <li><strong>${entrypointsToUpdate.map((e) => e.value).join('</strong></li><li><strong>')}</li>
+          <li><strong>${entrypointsToUpdateWithManyTags.map((e) => e.value).join('</strong></li><li><strong>')}</strong></li>
+        </ul>`;
+    }
+
+    if (entrypointsToUpdateWithOneTag.length === 1) {
+      entrypointsInfoMessage += `<br>The <strong>${entrypointsToUpdateWithOneTag[0].value}</strong> entrypoint will be deleted as it only used this tag.`;
+    } else if (entrypointsToUpdateWithOneTag.length > 1) {
+      entrypointsInfoMessage += `
+        <br>All these entrypoints will be deleted as they were only using this tag:
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]:
```suggestion
        <br>The following entrypoints will be deleted as they are only using this tag:
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts [hunk 4]
```diff
@@ -210,14 +210,27 @@ export class OrgSettingsTagsComponent implements OnInit, OnDestroy {
...
       entrypointsInfoMessage = `
         <br>The tag will be removed from all these entrypoints:
         <ul>
-          <li><strong>${entrypointsToUpdate.map((e) => e.value).join('</strong></li><li><strong>')}</li>
+          <li><strong>${entrypointsToUpdateWithManyTags.map((e) => e.value).join('</strong></li><li><strong>')}</strong></li>
+        </ul>`;
+    }
+
+    if (entrypointsToUpdateWithOneTag.length === 1) {
+      entrypointsInfoMessage += `<br>The <strong>${entrypointsToUpdateWithOneTag[0].value}</strong> entrypoint will be deleted as it only used this tag.`;
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]:
```suggestion
      entrypointsInfoMessage += `<br>The <strong>${entrypointsToUpdateWithOneTag[0].value}</strong> entrypoint will be deleted as it is only using this tag.`;
```

**Comment by ThibaudAV** [comment]
I have "mixed feelings" about SonarCloud Quality 🤔  
Is really necessary to do the tests for all these cases or not that's the question. 
maybe just one to test `of()` ligne 💭  🤷‍♂️

File: gravitee-apim-console-webui/src/entities/entrypoint/entrypoint.fixture.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ export function fakeEntrypoint(attributes?: Partial<Entrypoint>): Entrypoint {
   const base: Entrypoint = {
     id: 'da98cab3-ff94-4c68-98ca-b3ff949c6896',
     value: 'https://googl.co',
-    tags: ['internal'],
+    tags: ['external'],
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/entities/entrypoint/entrypoint.fixture.ts:22]: Mazette

**Review by gaetanmaisse — APPROVED** [review]

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]: 😮  what have I done here 😵‍💫

File: gravitee-apim-console-webui/src/entities/entrypoint/entrypoint.fixture.ts (hunk 5 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/entities/entrypoint/entrypoint.fixture.ts:22]: to make the fixtures consistent 🤷‍♂️

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-dwukwxjout.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-wcdekddswu.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags-par2/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Pull request merged**